File: necform/src/main/java/org/sid/necform/service/DocumentService.java
package org.sid.necform.service;

import org.sid.necform.dto.request.CreateDocumentRequest;
import org.sid.necform.dto.response.DocumentResponse;

import java.util.List;
import java.util.UUID;

public interface DocumentService {

    DocumentResponse create(
            CreateDocumentRequest request
    );

    List<DocumentResponse> getAll();

    DocumentResponse getById(
            UUID id
    );

    void delete(
            UUID id
    );

}

File: necform-front/src/app/core/services/document.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { 
  Document, 
  CreateDocument 
} from '../../shared/models/document.model';
import { ApiService } from './api.service';

@Injectable({
  providedIn: 'root'
})
export class DocumentService {
  private readonly endpoint: string;

  constructor(
    private http: HttpClient,
    private apiService: ApiService
  ) {
    this.endpoint = `${this.apiService.getApiUrl()}/documents`;
  }

  create(request: CreateDocument): Observable<Document> {
    return this.http.post<Document>(this.endpoint, request);
  }

  getAll(): Observable<Document[]> {
    return this.http.get<Document[]>(this.endpoint);
  }

  getById(id: string): Observable<Document> {
    return this.http.get<Document>(`${this.endpoint}/${id}`);
  }

  delete(id: string): Observable<void> {
    return this.http.delete<void>(`${this.endpoint}/${id}`);
  }
}
